crates/bevy_hierarchy/src/valid_parent_check_plugin.rs

```diff
@@ -53,7 +53,7 @@ impl<T> Default for ReportHierarchyIssue<T> {
 /// (See B0004 explanation linked in warning message)
 pub fn check_hierarchy_component_has_valid_parent<T: Component>(
     parent_query: Query<
-        (Entity, &Parent, Option<&bevy_core::Name>),
```

Comment:
Not necessary for this PR, but we could replace the `Option<&Name>` + `Entity` with `NameOrEntity`

Reply:
Yeah I agree, I'll leave it off just to keep things less controversial, but it'd be a nice little follow-up.